src/clr-angular/modal/modal.spec.ts

```diff
@@ -206,6 +223,21 @@ describe('Modal', () => {
     })
   );
 
+  it(
+    'should not be closed on backdrop click by default',
+    fakeAsync(() => {
+      const defaultsFixture = TestBed.createComponent(TestDefaultsComponent);
+      defaultsFixture.detectChanges();
+      compiled = defaultsFixture.nativeElement;
+
+      const backdrop: HTMLElement = compiled.querySelector('.modal-backdrop');
+
+      backdrop.click();
+      flushAndExpectOpen(defaultsFixture, true);
```

Comment:
I think we should trigger detect changes before this expect.

Reply:
That's what the `flushAndExpectOpen` function is for. It detects changes, waits for animations to finish, and _then_ checks the result.

Reply:
It is the first thing that happens inside `flushAndExpectOpen` helper function.
It wraps several lifecycle calls, `detectChanges` being one of them. The `tick()` for the fakeAsync is also encapsulated in the `flushAndExpectOpen`.

Reply:
ah gotcha.